tests/integ/modin/frame/test_aggregate.py

```diff
@@ -162,6 +172,95 @@ def test_agg_dup_col(numeric_native_df, func, expected_union_count):
         eval_snowpark_pandas_result(snow_df, numeric_native_df, func)
 
 
+class TestNamedAggDupColPandasFails:
+    def test_single_agg_on_dup_column(self, numeric_native_df):
+        # rename to have duplicated column
+        numeric_native_df.columns = ["A", "B", "A"]
+        snow_df = pd.DataFrame(numeric_native_df)
+
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Data must be 1-dimensional, got ndarray of shape (1, 2) instead"
+            ),
+        ):
+            numeric_native_df.agg(x=("A", "min"))
+
+        result_df = native_pd.DataFrame(
+            [[0, np.nan], [np.nan, 2.0]], columns=["A", "A"], index=["x", "x"]
+        )
+
+        with SqlCounter(query_count=1, union_count=1):
+            assert_snowpark_pandas_equal_to_pandas(
+                snow_df.agg(x=("A", "min")), result_df
+            )
+
+    @pytest.mark.parametrize("order", [True, False], ids=["dup_first", "dup_last"])
+    def test_single_agg_on_dup_and_non_dup_columns(self, numeric_native_df, order):
+        # rename to have duplicated column
+        numeric_native_df.columns = ["A", "B", "A"]
+        snow_df = pd.DataFrame(numeric_native_df)
+        with pytest.raises(
+            ValueError,
+            match=re.escape(
+                "Data must be 1-dimensional, got ndarray of shape (2, 2) instead"
+            ),
+        ):
+            numeric_native_df.agg(x=("A", "min"), y=("B", "min"))
+
+        if order:
+            kwargs = {"x": ("A", "min"), "y": ("B", "min")}
+        else:
+            kwargs = {"x": ("B", "min"), "y": ("A", "min")}
+        result_df = native_pd.DataFrame(
+            [[0, np.nan, np.nan], [np.nan, 2.0, np.nan], [np.nan, np.nan, 0]],
+            columns=["A", "A", "B"],
+            index=["x", "x", "y"] if order else ["y", "y", "x"],
+        )
+        if not order:
+            b_col = result_df.pop("B")
+            result_df.insert(0, "B", b_col)
+            result_df = result_df.sort_index()
+
+        with SqlCounter(query_count=1, union_count=2):
+            assert_snowpark_pandas_equal_to_pandas(
+                snow_df.agg(**kwargs),
+                result_df,
+            )
+
+    def test_multiple_agg_on_only_dup_columns(self, numeric_native_df):
+        # rename to have duplicated column
```

Comment:
is there a test with duplicated result lable like x = ('A', 'min'), x = ('B', 'max'), that should also error even if the no duplication in original df.

Reply:
That is a syntax error in python, so it should be intercepted by the python layer. See relevant comment here: https://github.com/snowflakedb/snowpark-python/pull/1652#discussion_r1612427237. To summarize, I'm not sure it makes sense to add a negative test for a python syntax error.